source/extensions/common/http_wasm/guest.cc

```diff
@@ -0,0 +1,431 @@
+#include "envoy/server/lifecycle_notifier.h"
+#include "source/common/config/datasource.h"
+#include "source/extensions/common/http_wasm/guest.h"
+#include "source/extensions/common/http_wasm/v8/v8.h"
+#include "source/extensions/common/http_wasm/exports.h"
+#include "source/extensions/common/http_wasm/vm_runtime.h"
+//#include "vm.h"
+#include <openssl/sha.h>
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace HttpWasm {
+
+namespace {
+// Map from Wasm Key to the local Wasm instance.
+thread_local std::unordered_map<std::string, std::weak_ptr<InitializedGuestHandle>>
+    local_initialized_guests;
+const std::string INLINE_STRING = "<inline>";
+
+inline Guest* getGuest(GuestHandleSharedPtr& guest_handle) { return guest_handle->guest().get(); }
+
+std::vector<uint8_t> Sha256(const std::vector<std::string_view>& parts) {
+  uint8_t sha256[SHA256_DIGEST_LENGTH];
+  SHA256_CTX sha_ctx;
+  SHA256_Init(&sha_ctx);
+  for (auto part : parts) {
+    SHA256_Update(&sha_ctx, part.data(), part.size());
+  }
+  SHA256_Final(sha256, &sha_ctx);
+  return std::vector<uint8_t>(std::begin(sha256), std::end(sha256));
+}
+std::string BytesToHex(const std::vector<uint8_t>& bytes) {
+  static const char* const hex = "0123456789ABCDEF";
+  std::string result;
+  result.reserve(bytes.size() * 2);
+  for (auto byte : bytes) {
+    result.push_back(hex[byte >> 4]);
+    result.push_back(hex[byte & 0xf]);
+  }
+  return result;
+}
+
+} // namespace
+
+std::string makeVmKey(std::string_view vm_id, std::string_view vm_configuration,
+                      std::string_view code) {
+  return BytesToHex(Sha256({vm_id, vm_configuration, code}));
+}
```

Comment:
what is a vm_configuration and vm_id in this context? let's use precise wording for clarity

Reply:
removed this function